administration/src/bp-modules/stores/StoresCSVInput.tsx

```diff
@@ -0,0 +1,147 @@
+import { useAppToaster } from '../AppToaster'
+import FileInputStateIcon, { FileInputStateType } from '../FileInputStateIcon'
+import { AcceptingStoreEntry } from './AcceptingStoreEntry'
+import { StoreData } from './StoresImportController'
+import StoresRequirementsText from './StoresRequirementsText'
+
+const StoreImportInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const InputContainer = styled(NonIdealState)`
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  justify-content: center;
+`
+type StoresCsvInputProps = {
+  setAcceptingStores: (store: AcceptingStoreEntry[]) => void
+  fields: StoreFieldConfig[]
+}
+
+export const FILE_SIZE_LIMIT_MEGA_BYTES = 2
+const defaultExtensionsByMIMEType = {
+  'text/csv': '.csv',
+}
+const FILE_SIZE_LIMIT_BYTES = FILE_SIZE_LIMIT_MEGA_BYTES * 1000 * 1000
+
+const lineToStoreEntry = (line: string[], headers: string[], fields: StoreFieldConfig[]): AcceptingStoreEntry => {
+  let storeData: StoreData = {}
+  line.forEach((entry, index) => {
+    const columnName = headers[index]
+    // TODO 1570 get geodata if no coordinates available
+    storeData = { ...storeData, [columnName]: entry }
+  })
+  return new AcceptingStoreEntry(storeData, fields)
+}
+
+const StoresCsvInput = ({ setAcceptingStores, fields }: StoresCsvInputProps): ReactElement => {
+  const [inputState, setInputState] = useState<FileInputStateType>('idle')
+  const fileInput = useRef<HTMLInputElement>(null)
+  const appToaster = useAppToaster()
+  const headers = fields.map(field => field.name)
+
+  const showInputError = useCallback(
+    (message: string) => {
+      appToaster?.show({ intent: 'danger', message })
+      setInputState('error')
+      if (!fileInput.current) return
+      fileInput.current.value = ''
+    },
+    [appToaster]
+  )
+
+  const onLoadEnd = useCallback(
+    (event: ProgressEvent<FileReader>) => {
+      const content = event.target?.result as string
+      let lines: [][] = []
+      try {
+        lines = parse(content, {
+          delimiter: ',',
+          encoding: 'utf-8',
+        })
+      } catch {
+        showInputError('Die Datei hat ein ungültiges Spaltenformat und kann nicht gelesen werden.')
+        return
+      }
+      const numberOfColumns = lines[0]?.length
+
+      if (!numberOfColumns) {
+        showInputError('Die gewählte Datei ist leer.')
+        return
+      }
+
+      if (lines.length < 2) {
+        showInputError('Die Datei muss mindestens einen Eintrag enthalten.')
+        return
+      }
+
+
+      if (!lines.every((line: string[]) => line.length === numberOfColumns)) {
+        showInputError('Keine gültige CSV Datei. Nicht jede Reihe enthält gleich viele Elemente.')
+        return
+      }
+
+      const csvHeader = lines.shift() ?? []
+
+      if (csvHeader.toString() !== headers.toString()) {
+        showInputError(`Die erforderlichen Spalten sind nicht vorhanden oder nicht in der richtigen Reihenfolge.`)
```

Comment:
This error is also currently shown when all the required columns are there and the order is correct, but there is an extra column added.
So that kind of doesn't match the error description.
But that's quite nit-picky so up to you whether to fix that

Reply:
added a separate check and test for that